### DISCUSSION TITLE
livecheck not working anymore

### DISCUSSION THREAD

**Discussion by ChristianKraemer** [Homebrew/discussions#4131] [Everyday usage] [answered]
### Output of `brew config`

```shell
HOMEBREW_VERSION: 3.6.20-68-gaa7f793
ORIGIN: https://github.com/Homebrew/brew
HEAD: aa7f7934ef2148c4c3f6df0071bd0ed7a84ee95a
Last commit: 2 hours ago
Core tap ORIGIN: https://github.com/Homebrew/homebrew-core
Core tap HEAD: c21b2c69a048732cf8b9507dfebd5b94ee6a7d0f
Core tap last commit: 51 minutes ago
Core tap branch: master
HOMEBREW_PREFIX: /usr/local
HOMEBREW_CASK_OPTS: []
HOMEBREW_DISPLAY: /private/tmp/com.apple.launchd.XdX1yULjpA/org.xquartz:0
HOMEBREW_GITHUB_API_TOKEN: set
HOMEBREW_INSTALL_FROM_API: set
HOMEBREW_MAKE_JOBS: 4
HOMEBREW_NO_ENV_HINTS: set
Homebrew Ruby: 2.6.10 => /System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/bin/ruby
CPU: quad-core 64-bit kabylake
Clang: 14.0.0 build 1400
Git: 2.39.1 => /usr/local/bin/git
Curl: 7.86.0 => /usr/bin/curl
macOS: 13.2-x86_64
CLT: 14.2.0.0.1.1668646533
Xcode: N/A
```

### Output of `brew doctor`

```shell
Your system is ready to brew.
```

### Description of issue

Updated Ventura (and brew - didn't look at the output of 'brew up').
Whe I run today "brew lc --newer-only --installed --quiet --full-name" I get:

Error: This command cannot be run while HOMEBREW_INSTALL_FROM_API is set!

Output of 'env | grep  HOMEBREW' is:

HOMEBREW_GITHUB_API_TOKEN=DoesNotMatter
HOMEBREW_PREFIX=/usr/local
HOMEBREW_CELLAR=/usr/local/Cellar
HOMEBREW_REPOSITORY=/usr/local/Homebrew
HOMEBREW_NO_ENV_HINTS=no

Where is "HOMEBREW_INSTALL_FROM_API" configured?

**Comment by ChristianKraemer** [comment]
Addition: Output of 'brew developer' is:

Developer mode is enabled.

**Comment by Rylan12** [comment, marked as answer]
The messaging here needs to be improved, thanks for reporting!

Set `HOMEBREW_NO_INSTALL_FROM_API` in your shell profile to opt out. Eventually `livecheck` will need to be updated to work with the new API install system, but for now this should fix the issue.

**Reply by elburro1887** [reply]
Shouldn't brew automatically fall-back to a install from source if there are no other options available? 

Or maybe ask the user `Install from source instead [Y/n]` ?

Same thing happened to me today with the `httpd` package and I wasted hours looking around for a solution, since I didn't know anything about this new API stuff.

**Reply by Rylan12** [reply]
> Shouldn't brew automatically fall-back to a install from source if there are no other options available?
>
> Or maybe ask the user `Install from source instead [Y/n]` ?

No, this isn't really possible. The point of an API install is that you don't need to have core definitions installed locally. So there isn't necessarily something we can fall back to in most cases.
